Catalogs.Application/Handlers/ItemHandlers/DeleteItemHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Catalogs.Application.Comands.ItemCommands;
+using MediatR;
+
+namespace Catalogs.Application.Handlers.ItemHandlers
+{
+    public sealed class DeleteItemHandler(IUnitOfWork unitOfWork) : IRequestHandler<DeleteItemComand>
+    {
+        private readonly IUnitOfWork _unitOfWork = unitOfWork;
+
+        public async Task Handle(DeleteItemComand comand, CancellationToken token)
+        {
+            var itemTypeExists = await _unitOfWork.ItemType.IsExistAsync(it => it.Id.Equals(comand.TypeId), token);
+
+            if (!itemTypeExists)
+            {
+                throw new NotFoundException(ItemTypeMessages.TypeNotFound);
+            }
+
+            var item = await _unitOfWork.Item.GetItemOfTypeByIdAsync(comand.TypeId, comand.Id, comand.TrackChanges, token);
+
```

Comment:
i think it's unnecessary. Immediately search for an item by its id and if there is no such thing, throw an exception